qt/widgets/instrumentview/src/InstrumentWidgetPickTab.cpp

```diff
@@ -464,6 +473,12 @@ void InstrumentWidgetPickTab::setSelectionType() {
     m_activeTool->setText("Tool: Pixel selection");
     surfaceMode = ProjectionSurface::PickSingleMode;
     plotType = DetectorPlotController::Single;
+  } else if (m_whole->isChecked()) {
+    m_selectionType = WholeInstrument;
+    m_activeTool->setText("Tool: Whole instrument selection");
+    surfaceMode = ProjectionSurface::MoveMode;
```

Comment:
Just a quick question. For the pixel/tube selection modes the surface mode is set to `Pick`something but here we set it to `MoveMode` - could you help me understand that?

Reply:
I did it because the plot only gets called in InstrumentWidgetPickTab::setSelectionType if the surface mode is EditShapeMode or MoveMode. See these lines a bit lower down:

```
    if (interactionMode == ProjectionSurface::EditShapeMode || interactionMode == ProjectionSurface::MoveMode) {
      updatePlotMultipleDetectors();
```

The PickXXX modes do call updatePlotMultipleDetectors via another route but that only happens when the user moves the cursor over a detector. That's probably not ideal for the new "whole instrument" button where the user probably expects the plot as soon as the new button is pressed

One side effect of this is that the user can navigate around with the the new "whole instrument" button pressed. Not sure if this is desirable or not

Reply:
Ah yes I see. Making it dependent on picking the cursor position makes no sense here because it would not change.

I think moving seems reasonable as picking certainly makes less sense.